openstf/pipeline/optimize_hyperparameters.py

```diff
@@ -22,13 +25,59 @@
 TIMEOUT: int = 200  # Stop study after the given number of second(s).
 TRAIN_HORIZONS: List[float] = [0.25, 24.0]
 
+# The dictonairy can be interpret the following way
+# Tuples; E.g. name:( (min, max), logarithmic); for intigers and floats
+#           or name:((option1, option2), logarithmic); picks option 2 for lgb and for all others option 1
+# List; E.g. name:[option1, option2, ...]; for categorical
+default_paramspace: dict = {
+    # General parameters
+    "learning_rate": ((0.01, 0.2), True),
+    "alpha": ((1e-8, 1.0), True),
+    "lambda": ((1e-8, 1.0), True),
+    "subsample": ((0.5, 0.99), False),
+    "min_child_weight": ((1, 6), False),
+    "max_depth": ((3, 10), False),
+    "colsample_bytree": ((0.5, 1.0), False),
+    "max_delta_step": ((1, 10), False),
+}
+# Important parameters, model specific
+# XGB specific
+xgb_paramspace: dict = {
+    "gamma": ((1e-8, 1.0), True),
+    "booster": ["gbtree", "dart"],
+    # , "gblinear" gives warnings because it doesn't use { colsample_bytree, gamma, max_delta_step, max_depth, min_child_weight, subsample }
+}
+
+# LGB specific
+lgb_paramspace: dict = {
+    "num_leaves": ((16, 62), False),
+    "boosting_type": ["gbdt", "dart", "rf"],
+    "tree_learner": ["serial", "feature", "data", "voting"],
+    "n_estimators": ((50, 150), False),
+    "min_split_gain": ((1e-8, 1), True),
+    "subsample_freq": ((1, 10), False),
+}
+
+
+def get_relevant_model_paramspace(
```

Comment:
Let try to be a bit more consistent with our naming. It looks like we use various terms for the same things? trial_params, model_params, hyper_params, relevant_model_params. We are probably talking about 1-2 things but we have more names than that it looks.